[Util] [util_for_PHP.c:145] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::shutdown'. argsCount: 0.

with today ext with 7.3, get this error:

```sh
2021-07-31 18:26:17.976446+08:00 [ERROR]    [Util] [util_for_PHP.c:145] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::bootstrap'. argsCount: 2.
Configuration File (php.ini) Path: /usr/local/etc/php
Loaded Configuration File:         /usr/local/etc/php/php.ini

2021-07-31 18:26:17.980848+08:00 [ERROR]    [Util] [util_for_PHP.c:145] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::shutdown'. argsCount: 0.
```

I have encountered a similar error, not sure if they are related. I'm installing v1.3 from the package provided [here](https://github.com/elastic/apm-agent-php/releases/download/v1.3/apm-agent-php_1.3_all.deb). It's a container setup and in one project everything works well, in another built and run in exactly the same way I get the error. Both on PHP 7.4.23.

```
2021-09-03 20:35:45.938158+00:00 [ERROR]    [Util] [util_for_PHP.c:145] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::interceptedCallPostHook'. argsCount: 2.
2021-09-03 20:35:45.938254+00:00 [ERROR]    [C-to-PHP] [tracer_PHP_part.c:213] [tracerPhpPartInterceptedCallPostHook] Exiting: resultCode: resultFailure (4)
```

> I have encountered a similar error, not sure if they are related. I'm installing v1.3 from the package provided [here](https://github.com/elastic/apm-agent-php/releases/download/v1.3/apm-agent-php_1.3_all.deb). It's a container setup and in one project everything works well, in another built and run in exactly the same way I get the error. Both on PHP 7.4.23.
> 
> ```
> 2021-09-03 20:35:45.938158+00:00 [ERROR]    [Util] [util_for_PHP.c:145] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::interceptedCallPostHook'. argsCount: 2.
> 2021-09-03 20:35:45.938254+00:00 [ERROR]    [C-to-PHP] [tracer_PHP_part.c:213] [tracerPhpPartInterceptedCallPostHook] Exiting: resultCode: resultFailure (4)
> ```

Same error here:

`Sep 21 15:29:03 ae35945ea9d9 apache2: Elastic APM PHP Tracer [PID: 1200] 2021-09-21 15:29:03.917762+00:00 [ERROR]    [Util] [util_for_PHP.c:146] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::interceptedCallPostHook'. argsCount: 2.
Sep 21 15:29:03 ae35945ea9d9 apache2: Elastic APM PHP Tracer [PID: 1200] 2021-09-21 15:29:03.917917+00:00 [ERROR]    [C-to-PHP] [tracer_PHP_part.c:214] [tracerPhpPartInterceptedCallPostHook] Exiting: resultCode: resultFailure (4)
`

Running on a dockericed environment

Hi, i have the same following error as above. Even though the errors, request in is made in background.
```
2022-04-28 22:44:41.166475+00:00 [PID: 13] [TID: 13] [ERROR]    [Util] [util_for_PHP.c:189] [callPhpFunction] call_user_function failed. Return value: -1. PHP function name: `\Elastic\Apm\Impl\AutoInstrument\PhpPartFacade::interceptedCallPostHook'. argsCount: 2.
2022-04-28 22:44:41.166904+00:00 [PID: 13] [TID: 13] [ERROR]    [Util] [util_for_PHP.c:202] [callPhpFunction] Exiting...; resultCode: resultFailure (4);
2022-04-28 22:44:41.167051+00:00 [PID: 13] [TID: 13] [ERROR]    [C-to-PHP] [tracer_PHP_part.c:208] [tracerPhpPartInterceptedCallPostHook] Exiting...; resultCode: resultFailure (4); dbgInterceptRegistrationId: 9; interceptedCallRetValOrThrown type: 2.
```

PHP Apm agent runs in Docker, version 1.5.

--- a/src/ext/backend_comm.c
+++ b/src/ext/backend_comm.c
@@ -206,14 +206,6 @@ ResultCode initConnectionData( const ConfigSnapshot* config, ConnectionData* con
          * @link https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYPEER.html
          */
         ELASTIC_APM_CURL_EASY_SETOPT( connectionData->curlHandle, CURLOPT_SSL_VERIFYPEER, 0L );
-
-        /**
-         * This option determines whether libcurl verifies the status of the server cert using the "Certificate Status Request" TLS extension (aka. OCSP stapling).
-         * Note that if this option is enabled but the server does not support the TLS extension, the verification will fail.
-         *
-         * @link https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYSTATUS.html
-         */
-        ELASTIC_APM_CURL_EASY_SETOPT( connectionData->curlHandle, CURLOPT_SSL_VERIFYSTATUS, 0L );
     }
 
     // Authorization with API key or secret token if present

--- a/src/ext/backend_comm.c
+++ b/src/ext/backend_comm.c
@@ -185,9 +185,35 @@ ResultCode initConnectionData( const ConfigSnapshot* config, ConnectionData* con
 
     if ( ! config->verifyServerCert )
     {
-        ELASTIC_APM_LOG_DEBUG( "verify_server_cert configuration option is set to false"
-                               " - disabling SSL/TLS certificate verification for communication with APM Server..." );
+        ELASTIC_APM_LOG_DEBUG( "verify_server_cert configuration option is set to false - disabling SSL/TLS certificate verification for communication with APM Server..." );
+        /**
+         * This option determines whether libcurl verifies that the server cert is for the server it is known as.
+         * When negotiating TLS and SSL connections, the server sends a certificate indicating its identity.
+         * When CURLOPT_SSL_VERIFYHOST is 2, that certificate must indicate that the server is the server to which you meant to connect, or the connection fails.
+         * Simply put, it means it has to have the same name in the certificate as is in the URL you operate against.
+         * When the verify value is 0, the connection succeeds regardless of the names in the certificate.
+         *
+         * @link https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYHOST.html
+         */
+        ELASTIC_APM_CURL_EASY_SETOPT( connectionData->curlHandle, CURLOPT_SSL_VERIFYHOST, 0L );
+
+        /**
+         * This option determines whether curl verifies the authenticity of the peer's certificate. A value of 1 means curl verifies; 0 (zero) means it does not.
+         * Authenticating the certificate is not enough to be sure about the server. You typically also want to ensure that the server is the server you mean to be talking to.
+         * Use CURLOPT_SSL_VERIFYHOST for that.
+         * The check that the host name in the certificate is valid for the host name you are connecting to is done independently of the CURLOPT_SSL_VERIFYPEER option.
+         *
+         * @link https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYPEER.html
+         */
         ELASTIC_APM_CURL_EASY_SETOPT( connectionData->curlHandle, CURLOPT_SSL_VERIFYPEER, 0L );
+
+        /**
+         * This option determines whether libcurl verifies the status of the server cert using the "Certificate Status Request" TLS extension (aka. OCSP stapling).
+         * Note that if this option is enabled but the server does not support the TLS extension, the verification will fail.
+         *
+         * @link https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYSTATUS.html
+         */
+        ELASTIC_APM_CURL_EASY_SETOPT( connectionData->curlHandle, CURLOPT_SSL_VERIFYSTATUS, 0L );
     }
 
     // Authorization with API key or secret token if present

